google/cloud/security/common/gcp_api/_base_repository.py

```diff
@@ -0,0 +1,695 @@
+
+        Args:
+          resource (str): The id of the resource to query.
+          fields (str): Fields to include in the response - partial response.
+          max_results (int): Number of entries to include per page.
+          verb (str): The method to call on the API.
+          kwargs (dict): Optional additional arguments to pass to the query.
+
+        Yields:
+          dict: An API response containing one page of results.
+        """
+        assert isinstance(self, GCPRepository)
+
+        arguments = {'fields': fields,
+                     self._max_results_field: max_results}
+
+        if self._list_key_field and resource:
+            arguments[self._list_key_field] = resource
+
+        if kwargs:
+            arguments.update(kwargs)
+
+        if self._request_supports_pagination(verb):
+            for resp in self.execute_paged_query(verb=verb,
+                                                 verb_arguments=arguments):
+                yield resp
+        else:
+            # Some API list() methods are not actually paginated.
+            del arguments[self._max_results_field]
+            yield self.execute_query(verb=verb, verb_arguments=arguments)
+
+
+class AggregatedListQueryMixin(ListQueryMixin):
+    """Mixin that implements a Paged List and AggregatedList query."""
+
+    def aggregated_list(self, resource=None, fields=None, max_results=None,
+                        verb='aggregatedList', **kwargs):
+        """List all subresource entities of a given resource.
+
+        Args:
+          resource (str): The id of the resource to query.
+          fields (str): Fields to include in the response - partial response.
+          max_results (int): Number of entries to include per page.
+          verb (str): The method to call on the API.
+          kwargs (dict): Optional additional arguments to pass to the query.
+
+        Returns:
+          iterator: An iterator of API responses by page.
+        """
+        return super(AggregatedListQueryMixin, self).list(
+            resource, fields, max_results, verb, **kwargs)
+
+
+class GetQueryMixin(object):
+    """Mixin that implements Get query."""
+
+    def get(self, resource, target=None, fields=None, verb='get', **kwargs):
+        """Get API entity.
+
+        Args:
+          resource (str): The id of the resource to query.
+          target (str):  Name of the entity to fetch.
+          fields (str): Fields to include in the response - partial response.
+          verb (str): The method to call on the API.
+          kwargs (dict): Optional additional arguments to pass to the query.
+
+        Returns:
+          dict: GCE response.
+
+        Raises:
+          errors.HttpError: When attempting to get a non-existent entity.
+           ex: HttpError 404 when requesting ... returned
+            "The resource '...' was not found"
+        """
+        assert isinstance(self, GCPRepository)
+        assert bool(self._get_key_field)
```

Comment:
:) The reason for the assert on GCPRepository was to get PyCharm figure out that self is GCPRepository and  provide proper code completion support.
It appears that this could as easily be achieved via a docstring for self.
self (GCPRepository): GCPRepository.

In other words, we should probably remove these asserts; Asserts on types are typically frowned upon :(